otherlibs/systhreads/Makefile

```diff
@@ -13,96 +13,127 @@
 #*                                                                        *
 #**************************************************************************
 
-include ../../config/Makefile
-CAMLRUN ?= ../../boot/ocamlrun
-CAMLYACC ?= ../../boot/ocamlyacc
-
 ROOTDIR=../..
-CAMLC=$(CAMLRUN) $(ROOTDIR)/ocamlc -nostdlib \
-      -I $(ROOTDIR)/stdlib -I $(ROOTDIR)/otherlibs/unix
-CAMLOPT=$(CAMLRUN) $(ROOTDIR)/ocamlopt -nostdlib \
-        -I $(ROOTDIR)/stdlib -I $(ROOTDIR)/otherlibs/unix
-MKLIB=$(CAMLRUN) ../../tools/ocamlmklib
+
+include $(ROOTDIR)/config/Makefile
+
+CAMLRUN ?= $(ROOTDIR)/boot/ocamlrun
+CAMLYACC ?= $(ROOTDIR)/boot/ocamlyacc
+
+LIBS = -nostdlib -I $(ROOTDIR)/stdlib -I $(ROOTDIR)/otherlibs/$(UNIXLIB)
+
+CAMLC=$(CAMLRUN) $(ROOTDIR)/ocamlc $(LIBS)
+CAMLOPT=$(CAMLRUN) $(ROOTDIR)/ocamlopt $(LIBS)
+MKLIB=$(CAMLRUN) $(ROOTDIR)/tools/ocamlmklib
 COMPFLAGS=-w +33..39 -warn-error A -g -bin-annot -safe-string
 ifeq "$(FLAMBDA)" "true"
 OPTCOMPFLAGS=-O3
 else
 OPTCOMPFLAGS=
 endif
 
-BYTECODE_C_OBJS=st_stubs_b.o
-NATIVECODE_C_OBJS=st_stubs_n.o
+LIBNAME=threads
+
+ifeq "$(UNIX_OR_WIN32)" "unix"
+HEADER = st_posix.h
+else # Windows
+HEADER = st_win32.h
+endif
+
+BYTECODE_C_OBJS=st_stubs_b.$(O)
+NATIVECODE_C_OBJS=st_stubs_n.$(O)
+
+THREADS_SOURCES := thread.ml mutex.ml condition.ml event.ml threadUnix.ml
```

Comment:
Alain Frisch (2016/08/30 15:10 -0700):
> >  COMPFLAGS=-w +33..39 -warn-error A -g -bin-annot -safe-string
> >  ifeq "$(FLAMBDA)" "true"
> >  OPTCOMPFLAGS=-O3
> >  else
> >  OPTCOMPFLAGS=
> >  endif
> >  
> > -BYTECODE_C_OBJS=st_stubs_b.o
> > -NATIVECODE_C_OBJS=st_stubs_n.o
> > +LIBNAME=threads
> > +
> > +ifeq "$(UNIX_OR_WIN32)" "unix"
> > +HEADER = st_posix.h
> > +else # Windows
> > +HEADER = st_win32.h
> > +endif
> > +
> > +BYTECODE_C_OBJS=st_stubs_b.$(O)
> > +NATIVECODE_C_OBJS=st_stubs_n.$(O)
> > +
> > +THREADS_SOURCES := thread.ml mutex.ml condition.ml event.ml threadUnix.ml
> 
> Why are you using `:=` (and also below)?

It's true that it may be better to be consistent.
That being said, I believe it is legitimate to use := in these places
and it would even be better to use it in other places, too.

See section 6.2 The Two Flavors of Variables in the make manual.

For the moment I replaced := by = everywhere, but I think in the long
run it may be better to use := where it makes sense, i.e. in a lot of
places, IMO.